src/reducers/sources.js

```diff
@@ -236,6 +244,16 @@ function updateSourceUrl(state: SourcesState, source: Source) {
   }
 }
 
+function updatePlainUrl(state: SourcesState, source: Source) {
+  // If we've seen this URL before, it will be in state.urls, and we shouldn't
+  // add it to plainUrls a second time.
+  if (source.url && !state.urls[source.url]) {
```

Comment:
maybe we should use `hasPlainUrlsForUrl` as well just so we don't make assumptions :)